tracer/test/Datadog.Trace.Tests/Telemetry/Metrics/MetricTests.cs

```diff
@@ -34,6 +34,16 @@ public class MetricTests
         { "init_time", new() { "total", "component" } }, // we only send one of these
     };
 
+    /// <summary>
+    /// The tags of the following metrics are ignored due to the difficulty to expose the actual values (it has an special enum to avoid generator limitations)
+    /// </summary>
+    private static readonly List<string> IgnoreTagsPermutationValidation = new()
+    {
+        "event_created",
+        "event_finished",
+        "git_requests.settings_response",
+    };
+
```

Comment:
I will need your help with this I was trying with the permutations thing with no luck, this is related to the custom `enum` logic to avoid the changes to support more than 2 tags...

```csharp
    public static MetricTags.CIVisibilityTestingEventTypeWithCodeOwnerAndSupportedCiAndBenchmark? GetEventTypeWithCodeOwnerAndSupportedCiAndBenchmark(MetricTags.CIVisibilityTestingEventType eventType, bool isBenchmark)
        => eventType switch
        {
            MetricTags.CIVisibilityTestingEventType.Test => isBenchmark
                                                                ? MetricTags.CIVisibilityTestingEventTypeWithCodeOwnerAndSupportedCiAndBenchmark.Test_IsBenchmark
                                                                : MetricTags.CIVisibilityTestingEventTypeWithCodeOwnerAndSupportedCiAndBenchmark.Test,
            MetricTags.CIVisibilityTestingEventType.Suite => MetricTags.CIVisibilityTestingEventTypeWithCodeOwnerAndSupportedCiAndBenchmark.Suite,
            MetricTags.CIVisibilityTestingEventType.Module => MetricTags.CIVisibilityTestingEventTypeWithCodeOwnerAndSupportedCiAndBenchmark.Module,
            MetricTags.CIVisibilityTestingEventType.Session => CIEnvironmentValues.Instance switch
            {
                { CodeOwners: not null, IsCI: true } => MetricTags.CIVisibilityTestingEventTypeWithCodeOwnerAndSupportedCiAndBenchmark.Session_HasCodeOwner_IsSupportedCi,
                { CodeOwners: not null, IsCI: false } => MetricTags.CIVisibilityTestingEventTypeWithCodeOwnerAndSupportedCiAndBenchmark.Session_HasCodeOwner_UnsupportedCi,
                { CodeOwners: null, IsCI: true } => MetricTags.CIVisibilityTestingEventTypeWithCodeOwnerAndSupportedCiAndBenchmark.Session_NoCodeOwner_IsSupportedCi,
                { CodeOwners: null, IsCI: false } => MetricTags.CIVisibilityTestingEventTypeWithCodeOwnerAndSupportedCiAndBenchmark.Session_NoCodeOwner_UnsupportedCi,
            },
            _ => null,
        };
```

and the settings response that includes the `string.Empty` value:

```csharp
    public enum CIVisibilityITRSettingsResponse
    {
        [Description("")] CoverageDisabled_ItrSkipDisabled,
        [Description("coverage_enabled")] CoverageEnabled_ItrSkipDisabled,
        [Description("itrskip_enabled")] CoverageDisabled_ItrSkipEnabled,
        [Description("coverage_enabled;itrskip_enabled")] CoverageEnabled_ItrSkipEnabled,
    }
```

Reply:
Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/4596/commits/db0de65eaf29330db687e7623eb223ff70dd5b82